extmod/modre.c

```diff
@@ -194,7 +194,7 @@ static void re_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t
     mp_printf(print, "<re %p>", self);
 }
 
-static mp_obj_t re_exec(bool is_anchored, uint n_args, const mp_obj_t *args) {
+static mp_obj_t mp_re_exec(bool is_anchored, uint n_args, const mp_obj_t *args) {
```

Comment:
I suggest calling it `re_exec_helper`, because that matches other naming in this project.

Also probably worth adding a comment along the lines of "this function can't be named re_exec because it may clash with system headers, eg on FreeBSD".